From eb6b3486faa7d8e2dc810cee8297ee160cfe6a5c Mon Sep 17 00:00:00 2001
From: AppaRao Puli <apparao.puli@intel.com>
Date: Thu, 6 Jul 2023 10:06:46 +0000
Subject: [PATCH] Fix for EventService subscription collection fail

With recent SNMP support for EventService is causing EventCollection
fecth with 500 - Internal server error. Some OEM's doesn't have SNMP
service to handle the subscriptions and this is not mandatory service.
So added code to ignore the SNMP service not found cases.

Tested:
GET on EventService/Subscriptions works as expected if SNMP service
doesn't exist.

Signed-off-by: AppaRao Puli <apparao.puli@intel.com>
---
 redfish-core/lib/event_service.hpp | 10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git a/redfish-core/lib/event_service.hpp b/redfish-core/lib/event_service.hpp
index 62e912b0..bd5fdd9d 100644
--- a/redfish-core/lib/event_service.hpp
+++ b/redfish-core/lib/event_service.hpp
@@ -192,14 +192,8 @@ inline void doSubscriptionCollection(
 {
     if (ec)
     {
-        if (ec.value() == EBADR)
-        {
-            // This is an optional process so just return if it isn't there
-            return;
-        }
-
-        BMCWEB_LOG_ERROR("D-Bus response error on GetManagedObjects {}", ec);
-        messages::internalError(asyncResp->res);
+        // This is an optional process so just return if it isn't there
+        BMCWEB_LOG_DEBUG("EventService: The SNMP service is not enabled");
         return;
     }
     nlohmann::json& memberArray = asyncResp->res.jsonValue["Members"];
-- 
2.25.1

